### PR TITLE
Porting babel-plugin-transform-for-of-as-array into transform-for-of as option

### PR DESCRIPTION
**Pull request by rajzshkr**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | #6816 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      | Yes
| Tests Added + Pass?      | Yes
| Documentation PR         | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

**Comment by rajzshkr** [comment]
Usage:-

```js
{
  "plugins": [["transform-for-of", {
    "forOfAsArray": true
    }]
  ]
}
```

File: packages/babel-plugin-transform-for-of/src/index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { template, types as t } from "@babel/core";
 
 export default function(api, options) {
-  const { loose } = options;
+  const { loose, forOfAsArray: isForOfAsArray } = options;
```
Comment (Andarist) [inline packages/babel-plugin-transform-for-of/src/index.js]: would prefer a little bit more descriptive name - `useForAsArray` or something and that name could be used both as the internal flag and the public plugin option

Comment (nicolo-ribaudo) [reply packages/babel-plugin-transform-for-of/src/index.js]: What about just `assmeArray`? After all, `for of` is already specified in the plugin name.

Comment (rajzshkr) [reply packages/babel-plugin-transform-for-of/src/index.js]: @nicolo-ribaudo assumeArray?

Comment (Andarist) [reply packages/babel-plugin-transform-for-of/src/index.js]: Yeah, `assumeArray` sounds like a great name for this.

Comment (rajzshkr) [reply packages/babel-plugin-transform-for-of/src/index.js]: yeah, I will update the PR.

**Comment by nicolo-ribaudo** [comment]
1. Since the `loose` and `assumeArray` options are not compatible, I think we should throw an error when both are `true`.
2. If `assumeArray` if `true`, we can avoid defining the [`buildForOfArray`](https://github.com/babel/babel/pull/6914/files#diff-55e984c93d4c177235e494c251a9eadeR9), [`buildForOfLoose`](https://github.com/babel/babel/pull/6914/files#diff-55e984c93d4c177235e494c251a9eadeR13) and [`buildForOf`](https://github.com/babel/babel/pull/6914/files#diff-55e984c93d4c177235e494c251a9eadeR31) templates by returing earlier.

File: packages/babel-plugin-transform-for-of/src/index.js [hunk 2]
```diff
@@ -1,7 +1,61 @@
 import { template, types as t } from "@babel/core";
 
 export default function(api, options) {
-  const { loose } = options;
+  const { loose, assumeArray } = options;
+
+  if (loose === true && assumeArray === true) {
+    throw new Error(
+      `Cannot use loose and assumeArray options together in babel-plugin-transform-for-of`,
```
Comment (existentialism) [inline packages/babel-plugin-transform-for-of/src/index.js]: Nit:

> The `loose` and `assumeArray` options cannot be used together in @babel/plugin-transform-for-of

File: packages/babel-plugin-transform-for-of/src/index.js [hunk 3]
```diff
@@ -107,49 +161,6 @@ export default function(api, options) {
     }
   }
 
-  return {
```
Comment (hzoo) [inline packages/babel-plugin-transform-for-of/src/index.js]: none of the code here is any different right? (just saying this part of the diff isn't necessary as clarification)

**Comment by hzoo** [comment]
We can test this by using it lol, but can be another PR https://github.com/babel/babel/blob/d8bbaaae0a132cf464c27126957bd66d74b2bf65/packages/babylon/package.json#L27

**Pull request merged**